web-app/src/app/screens/FeedSubmission/Form/index.tsx

```diff
@@ -129,18 +132,49 @@ export default function FeedSubmissionForm(): React.ReactElement {
     handleBack();
   };
 
-  const finalSubmit = (
+  const finalSubmit = async (
     partialFormData: Partial<FeedSubmissionFormFormInput>,
-  ): void => {
+  ): Promise<void> => {
     const finalData = { ...formData, ...partialFormData };
+    setIsSubmitLoading(true);
     setFormData(finalData);
-    // console.log('FINAL API CALL WITH', finalData);
-    // TODO: API call with finalData
-    // TODO: loading state of API call
-    // TODO: feed submitted page
-    handleNext();
+    try {
+      const auth = getAuth();
+      const requestBody = { ...finalData, userId: auth.currentUser?.uid ?? '' };
+      await submitNewFeedForm(requestBody);
+      handleNext();
+    } catch (error) {
+      setSubmitError(t('form.errorSubmitting'));
+    } finally {
+      setIsSubmitLoading(false);
+    }
   };
 
+  if (isSubmitLoading) {
+    return (
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'center',
+          flexWrap: 'wrap',
+          mt: 3,
+        }}
+      >
+        <CircularProgress />
+        <Typography
+          variant='h6'
+          sx={{ width: '100%', textAlign: 'center', mt: 2 }}
+        >
+          {t('form.submittingFeed')}
+        </Typography>
+      </Box>
+    );
+  }
+
+  if (submitError !== undefined) {
+    return <Typography>{submitError}</Typography>;
+  }
```

Comment:
currently if submitting the feed fails, it shows a basic error message "Submitting feed failed". Is there a more complex message or action we want to provide?